app/main.py

```diff
@@ -1 +1,42 @@
-# write your code here
+from __future__ import annotations
+import os
+from typing import Any
+
+
+class MoveFile:
+    def __init__(
+            self,
+            old_filename: str,
+            new_filename: str,
+            direction: str
+    ) -> None:
+        self.old_filename = old_filename
+        self.new_filename = new_filename
+        self.direction = direction
+
+    def __enter__(self) -> MoveFile:
+        self.new_filename = open(f"{self.direction}/{self.new_filename}", "w")
+        with open(self.old_filename, "r") as f:
+            content_to_copy = f.read()
+        self.new_filename.write(content_to_copy)
+        return self.new_filename
+
+    def __exit__(self, exc_type: Any, exc_val: Any, exc_tb: Any) -> None:
```

Comment:
https://adamj.eu/tech/2021/07/04/python-type-hints-how-to-type-a-context-manager/#class-based-context-managers
Provide annotations